dll/win32/shell32/lang/da-DK.rc

```diff
@@ -362,23 +362,22 @@ BEGIN
 END
 
 IDD_RUN_AS DIALOGEX 0, 0, 240, 190
-STYLE DS_SHELLFONT | WS_CHILD | WS_DISABLED | WS_CAPTION
+STYLE DS_SHELLFONT | DS_MODALFRAME | WS_POPUP | WS_CAPTION | WS_SYSMENU
 CAPTION "Run As"
 FONT 8, "MS Shell Dlg", 0, 0, 0x0
 BEGIN
-    LTEXT "Which user account do you want to use to run this program?", -1, 10, 20, 220, 20
-    CHECKBOX "Current User %s", 14000, 10, 45, 150, 10
-    LTEXT "Protect my computer and data from unauthorized program activity", -1, 25, 57, 200, 10, WS_DISABLED
-    CHECKBOX "This option can prevent computer viruses from harming your computer or personal data, but selecting it might cause the program to function improperly.", 14001, 25, 68, 200, 30, WS_DISABLED | BS_MULTILINE
-    CHECKBOX "The following user:", 14002, 10, 100, 90, 10
+    ICON IDI_SHELL_SECURITY, -1, 8, 8, 21, 20
```

Comment:
I do request changes before this PR can be merged!
You added ICON IDI_SHELL_SECURITY into some but not all translations (it is missing in de-De, ro-RO and most likely others as well). This is definitely not a good idea. Either add it everywhere, or add it nowhere.
If the position in relation to other controls would not be perfect in all languages yet, that might be tolerable.
But translators will never know that something is wrong with their specific translation if ICON IDI_SHELL_SECURITY is missing entirely for them.

Reply:
There is more wrong with the translated dialogs than just a missing icon. This dialog should have been rejected in its initial PR!

I don't want to be the one spending my time fixing them when they should not have required fixing in the first place. I can add a `/* FIXME: Missing ICON IDI_SHELL_SECURITY */` if that is acceptable to you?

Reply:
No. If you add ICON IDI_SHELL_SECURITY into one rc, then add it everywhere.
If you do not want to add it everywhere, then add it nowhere.

Same for your AUTOCHECKBOX.

Reply:
And if that seems too complicated for you, then just copy over the english dlg over all other languages and then just restore the translations for each control. That is also fine, especially if you consider the whole thing to be an entire mess regarding sync between rcs.

Reply:
Some of the translated dialogs already use different sizes for the text controls, I cannot just copy the English one and replace with translated strings.

If a complete English dialog and mostly working other dialogs is not acceptable then you can close this PR instead...

Reply:
"different sizes for the text controls. I cannot just copy the English one and replace with translated strings."

You can do that ofc. Might introduce a bit text-cutoff here and there. So what? Translators will see that and fix that afterwards like they always used to do. Either by shortening their text, or enlarging the controls.

But having the controls in different order each rc, or having specific controls missing entirely, that is definitely not acceptable.

It would be a pity, if you had to close your PR out of lazyness to properly take care of the rc changes that your PR does require. What I suggested is something that you can achieve in 20minutes.

But yes: closing your PR would be better than merging its current state.

Reply:
The thing is that "adding/removing and reordering" controls is not part of a translators job. Their job is solely to provide translations to strings, preferrably those that match the existing controls sizes. Or less preferrably: enlarging a control in their specific languages.

A translator should never have to open another 20 languages rc to fulfill the job for his language just to spot missing controls. He is usually not skilled with a compare-tool, and shouldn't have to be required to. But that is a skill that we can absolutely expect a developer to master. Especially when it is his controls that he is just adding right now.

Why was the past code shitty?
I have not checked the history, maybe the dlg dates back to ancient SVN times, where reviews were not a thing yet. I cannot say. Most likely it differs just in the controls dimensions though, and nothing else. That is not beautiful sure, but also no catastrophy.

So again: Make sure that all controls are present in all languages, and in the same order.
If you have to make tabula rasa for that by copying stuff over from en-US, that is fine.

**Wise** would it be to enlarge a few controls beforehand in en-US, that you think may have been chosen to have very restrictive dimensions historically in en-US.
**Wise**, but I did not even require you to go that far.

**Friendly** would it be, to restore the existing strings after copy over.
**Friendly**, but I would not even insist on that. If you are unfriendly you can also throw all the past work of the translators away.

Best would be to be **wise** and **friendly** though.

**What I do insist of though, is that the same controls with the same names and in the same order do exist in all rcs.**

Reply:
I expect a translator to look at the English dialog if they see a FIXME comment telling them to copy the icon control from the English dialog.

 > Especially when it is his controls that he is just adding right now.

I did not want to, I was forced to edit these dialogs. I just tried to be nice and make the English one great once I was there.

> Most likely it differs just in the controls dimensions though, and nothing else.

The original dialogs are unusable to put it mildly. Somebody just looked at the dialog on XP at some point and then from memory tried to recreate the thing without ever testing anything or looking at Windows. The only valuable thing here are the strings. The original dialog has a browse button for the password field FFS. How do you browse for a password?

> What I do insist of though, is that the same controls with the same names and in the same order do exist in all rcs.

Maybe you are also referring to the "Protect my computer" checkbox and text? I did not spend as much effort fixing them and as I explained in my initial PR text, this checkbox will never be enabled so it does not matter if it is unusable or if the checkbox id does not match. We have the benefit of a time machine here and know that this setting is [too strict](https://learn.microsoft.com/en-us/archive/blogs/aaron_margosis/running-restricted-what-does-the-protect-my-computer-option-mean) and applications don't work well with restricted tokens and the whole thing was replaced by UAC in Vista.

Reply:
> I expect a translator to look at the English dialog if they see a FIXME comment telling them to copy the icon control from the English dialog.

Copying a FIXME there is the exact same amount of work as copying the real thing there.
It is **your PR** that added new controls into some RCs, and reordered existing ones.
It is **your PR** that changed the types and names of specific controls.

Therefore it is the responsibility of **your PR** to reflect those changes into all RCs.
I really offered you many opportunities for taking the easy way out here.
The condition that **the same controls with the same names and types in the same order will exist in all rcs** after your PR is not negotiable.

Reply:
> Copying a FIXME there is the exact same amount of work as copying the real thing there.

It is not. A FIXME comment is me acknowledging that there is a problem/something missing but I'm not fixing it right now. Adding controls would require me to check that the icon does not conflict with the text and be responsible for any bugs it might cause.

If translators are only responsible for the strings, why do we even need all these .rc files? Can't the translations just exist as #defines in a header and you can just have one .rc that you #include multiple times? As in,  `CONTROL IDC_WHATEVER, RCTRANSLATESTRING(WHATEVER), 42, 42, 42, 42` and then the define would be `#define SWEDISH_WHATEVER "Bork bork"` etc.

Reply:
No they can't.

> Maybe you are also referring to...

I am referring to
ICON IDI_SHELL_SECURITY that **you added**.
Controls that **you changed** the type into AUTORADIOBUTTONs
Contorls that **you changed** the type into AUTOCHECKBOXs.
Controls which names **you changed** from numbers to identifyers.

All of that is done inconsistently between the RCs. And all of that must be harmonized between all RCs.
This is no historic mess, as you called it. Those are **your changes**.

The time we are discussing it here, is larger already than the time it would have needed you to fix that.

Reply:
> This is no historic mess, as you called it. Those are your changes.

The historic mess is that the original dialogs don't work, they might as well not be there, would have saved me a lot of time to just paste the English version everywhere instead.

The controls I changed to AUTO make it work better but it still works without it. I don't know how many times I have to explain the checkbox. The checkbox is always disabled, WS_DISABLED is set in the RC and I claim that this feature will never be implemented. I just made it into the control it is supposed to be in the English dialog for future reference.

If you want all dialogs to be exactly the same then they will all become English. At least the translated string will still be available in the history if somebody wants to fix their own language.

Reply:
> If you want all dialogs to be exactly the same then they will all become English. 

Like I said before: This is **one** of the options you have. It is neither **friendly** nor **wise** to perform like that (see above). But it is one of the options you have.
I would at least suggest you to do it **wise** then, if you are too lazy to do it **friendly**.

Reply:
I at least tried to follow [the guidelines](https://learn.microsoft.com/en-us/windows/win32/uxguide/vis-layout#recommended-sizing-and-spacing) while leaving plenty of room for longer translations.

Reply:
This topic has been resolved by copying over the english dlg. Thx.